x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_edit.tsx

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useCallback, useReducer, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiTitle,
+  EuiFlyoutHeader,
+  EuiFlyout,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlyoutBody,
+  EuiPortal,
+  EuiBetaBadge,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useAlertsContext } from '../../context/alerts_context';
+import { Alert, AlertAction, IErrorObject } from '../../../types';
+import { AlertForm, validateBaseProperties } from './alert_form';
+import { alertReducer } from './alert_reducer';
+import { updateAlert } from '../../lib/alert_api';
+
+interface AlertEditProps {
+  initialAlert: Alert;
+  editFlyoutVisible: boolean;
+  setEditFlyoutVisibility: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const AlertEdit = ({
+  initialAlert,
+  editFlyoutVisible,
+  setEditFlyoutVisibility,
+}: AlertEditProps) => {
+  const [{ alert }, dispatch] = useReducer(alertReducer, { alert: initialAlert });
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const {
+    reloadAlerts,
+    http,
+    toastNotifications,
+    alertTypeRegistry,
+    actionTypeRegistry,
+  } = useAlertsContext();
+
+  const closeFlyout = useCallback(() => {
+    setEditFlyoutVisibility(false);
+    setServerError(null);
+  }, [setEditFlyoutVisibility]);
+
+  const [serverError, setServerError] = useState<{
+    body: { message: string; error: string };
+  } | null>(null);
+
+  if (!editFlyoutVisible) {
+    return null;
+  }
+
+  const alertType = alertTypeRegistry.get(alert.alertTypeId);
+  if (!alertType) {
+    return null;
+  }
```

Comment:
This shouldn't be needed, as the registry can't return null.
If the alert type is missing - it throws.